pelican/readers.py

```diff
@@ -672,7 +672,7 @@ def parse_path_metadata(source_path, settings=None, process=None):
                     # .items() for py3k compat.
                     for k, v in match.groupdict().items():
                         k = k.lower()  # metadata must be lowercase
-                        if k not in metadata:
+                        if k not in metadata and v is not None:
```

Comment:
Not that big of a deal but I'd prefer switching the comparison order for performance reasons. Comparison to `None` would be faster and should short-circuit earlier:

```python
if v is not None and k not in metadata:
```